src/execution_plan/execution_plan_reduce_to_apply.c

```diff
@@ -0,0 +1,164 @@
+#include "execution_plan.h"
+#include "ops/ops.h"
+#include "../query_ctx.h"
+#include "../ast/ast_mock.h"
+#include "./optimizations/optimizer.h"
+
+static OpBase *_buildMatchBranch(ExecutionPlan *plan, const cypher_astnode_t *path) {
+	AST *ast = QueryCtx_GetAST();
+
+	rax *bound_vars = NULL;
+	const char **arguments = NULL;
+	if(plan->root) {
+		bound_vars = raxNew();
+		// Rather than cloning the record map, collect the bound variables along with their
+		// parser-generated constant strings.
+		ExecutionPlan_BoundVariables(plan->root, bound_vars);
+		// Prepare the variables for populating the Argument ops we will build.
+		arguments = ExecutionPlan_BuildArgumentModifiesArray(bound_vars);
+	}
+
+	// Initialize an ExecutionPlan that shares this plan's Record mapping.
+	ExecutionPlan *match_branch_plan = ExecutionPlan_NewEmptyExecutionPlan();
+	match_branch_plan->record_map = plan->record_map;
+
+	// If we have bound variables, build an Argument op that represents them.
+	if(arguments) match_branch_plan->root = NewArgumentOp(plan, arguments);
+
+	// Build a temporary AST holding a MATCH clause.
+	AST *match_branch_ast = AST_MockMatchPattern(ast, path);
+
+	ExecutionPlan_PopulateExecutionPlan(match_branch_plan, NULL);
+
+	AST_MockFree(match_branch_ast);
+	QueryCtx_SetAST(ast); // Reset the AST.
+
+	OpBase *branch_match_root = match_branch_plan->root;
+	ExecutionPlan_BindPlanToOps(branch_match_root, plan);
+	OpBase *argument = ExecutionPlan_LocateFirstOp(branch_match_root, OPType_ARGUMENT);
+	if(argument) {
+		if(argument->modifies) {
```

Comment:
I don't think the check on `op->modifies` is actually necessary, but if it is, combine with above `if`.

Reply:
removed, we should always have argument and argument's modifies